LLama/LLamaWeights.cs

```diff
@@ -71,6 +71,23 @@ private LLamaWeights(SafeLlamaModelHandle weights)
             Metadata = weights.ReadMetadata();
         }
 
+        /// <summary>
+        /// Create from a "shared" handle
+        /// </summary>
+        /// <param name="handle"></param>
+        /// <returns></returns>
+        public static LLamaWeights FromSafeModelHandle(SafeLlamaModelHandle handle)
+        {
+            var model = new LLamaWeights(handle);
```

Comment:
Rather than exposing this as a static constructor, would it make sense to expose it as an instance method on `LLamaWeights`? Something like `CreateShared`, which can be used to effectively clone the `LLamaWeights` object to a new one with shared ownership of the underlying `SafeLlamaModelHandle`?

Reply:
Note: if you did this then the cache could go back to storing `LLamaWeights` directly. Every item returned from the cache would just call `CreateShared` on the item in the cache.

Reply:
This has been done with the clone from metadata method